generator/src/main/java/com/scottlogic/deg/generator/validators/GenerationConfigValidator.java

```diff
@@ -21,4 +40,48 @@ public ValidationResult validateCommandLine(GenerationConfig config) {
 
         return validationResult;
     }
-}
\ No newline at end of file
+
+    public ValidationResult validateCommandLinePostProfile(Profile profile) {
+        ArrayList<String> errorMessages = new ArrayList<>();
+        ValidationResult validationResult = new ValidationResult(errorMessages);
+
+        if (configSource.shouldViolate()) {
+            checkViolationGenerateOutputTarget(errorMessages, outputTarget, profile.rules.size());
+        } else {
+            checkGenerateOutputTarget(errorMessages, outputTarget);
+        }
+
+        return validationResult;
+    }
+
+    private void checkGenerateOutputTarget(ArrayList<String> errorMessages,
+                                           OutputTarget outputTarget) {
+        if (outputTarget instanceof FileOutputTarget) {
```

Comment:
If we are checking here then the FileUtils definitely doesn't need to worry about OutputTargets. Either use FileOutputTarget or Path/File inputs